sdk/python/tests/components/test_python_op.py

```diff
@@ -705,6 +705,26 @@ def write_to_file_path(
             },
         )
 
+    def test_annotations_stripping(self):
+        import typing
+        import collections
+
+        MyFuncOutputs = typing.NamedTuple('Outputs', [('sum', int), ('product', int)])
+
+        class CustomType1:
+            pass
+
+        def my_func(
+            param1: CustomType1 = None,  # This caused failure previously
+            param2: collections.OrderedDict = None,  # This caused failure previously
```

Comment:
Just curious, does hint striping reserve the default value by any chance? It might be useful to reserve it sometimes when, for example, the custom type is just an alias of (Text|str)

Reply:
The default values should always be preserved. Only annotations are stripped.